src/dialogxml/dialogs/dialog.cpp

```diff
@@ -43,6 +43,8 @@ const short cDialog::BG_DARK = 5, cDialog::BG_LIGHT = 16;
 short cDialog::defaultBackground = cDialog::BG_DARK;
 cDialog* cDialog::topWindow = nullptr;
 void (*cDialog::redraw_everything)() = nullptr;
+std::function<sf::RenderWindow&()> cDialog::get_mini_map;
+bool* cDialog::map_visible_p = nullptr;
```

Comment:
I _really_ don't like all this minimap handling in `cDialog`. It has no business knowing that a minimap even exists.

Reply:
I agree, and I think I've come up with a better way. Instead of have a get_mini_map() delegate it should pass an onLostFocus and onGainedFocus delegate, so the game code can supply all of that logic.